apollo-client/src/main/java/com/ctrip/framework/apollo/util/yaml/YamlParser.java

```diff
@@ -64,7 +65,16 @@ public void process(Properties properties, Map<String, Object> map) {
    * Create the {@link Yaml} instance to use.
    */
   private Yaml createYaml() {
-    return new Yaml(new StrictMapAppenderConstructor());
+    StrictMapAppenderConstructor constructor = new StrictMapAppenderConstructor();
+    Representer representer = new Representer();
+    DumperOptions dumperOptions = new DumperOptions();
+    dumperOptions.setDefaultFlowStyle(representer.getDefaultFlowStyle());
+    dumperOptions.setDefaultScalarStyle(representer.getDefaultScalarStyle());
+    dumperOptions.setAllowReadOnlyProperties(representer.getPropertyUtils().isAllowReadOnlyProperties());
+    dumperOptions.setTimeZone(representer.getTimeZone());
+    LoaderOptions loadingConfig = new LoaderOptions();
+    loadingConfig.setAllowDuplicateKeys(false);
+    return new Yaml(constructor, representer, dumperOptions, loadingConfig);
```

Comment:
Now that LoaderOptions provide the `setAllowDuplicateKeys` option, we don't need the `StrictMapAppenderConstructor` anymore?

```suggestion
    LoaderOptions loadingConfig = new LoaderOptions();
    loadingConfig.setAllowDuplicateKeys(false);
    return new Yaml(new SafeConstructor(), new Representer(), new DumperOptions(), loadingConfig);
```

Reply:
BTW, why would we need to set the configs of `DumperOptions`?